grails-app/controllers/org/pih/warehouse/core/LocationController.groovy

```diff
@@ -121,6 +126,8 @@ class LocationController {
                     return
                 }
             } else {
+                // Refresh to avoid saving binded, not validated data to the persisted object
+                locationInstance.refresh()
```

Comment:
If you don't understand a purpose of it, I can explain it, but let me know before, as it might be a longer comment/discussion, but simply:
```groovy
locationInstance.properties = params
```
don't validate the data, so it assumes that they are correct.
If we do this at the beginning:
```groovy
def locationInstance = inventoryService.getLocation(params.id)
```
this `Location` object is already persisted in the Hibernate session, so whenever we change any of properties of a persisted object, we don't have to explicity do `save()` after, because the data is saved after the session ends.
What happened was that even though I was getting a `ValidationException` having `["NONE", "SUBMIT_REQUEST"]`, it got saved due to that, so before the session ends, I want to refresh the instance to its "old" version if any validation exception occurs.